src/transaction_signature_options/sign_with_private_key/mod.rs

```diff
@@ -46,25 +51,38 @@ impl SignPrivateKeyContext {
         let signer_secret_key: near_crypto::SecretKey = scope.signer_private_key.clone().into();
         let public_key: near_crypto::PublicKey = scope.signer_public_key.clone().into();
 
-        let rpc_query_response = network_config
-            .json_rpc_client()
-            .blocking_call_view_access_key(
-                &previous_context.prepopulated_transaction.signer_id,
-                &public_key,
-                near_primitives::types::BlockReference::latest(),
+        let (nonce, block_hash, block_height) = if previous_context.offline {
+            (
+                scope.nonce.unwrap(),
+                scope.block_hash.unwrap().0,
+                scope.block_height.unwrap(),
```

Comment:
Return `Err` instead of crashing the application.

```rust
scope.nonce.wrap_err("Nonce is required to sign a transaction in offline mode")?,
```

Reply:
Done